go/protocol/keybase1/extras.go

```diff
@@ -2337,9 +2351,23 @@ func (r TeamRole) IsOrAbove(min TeamRole) bool {
 	case TeamRole_NONE:
 		return min == TeamRole_NONE
 	case TeamRole_RESTRICTEDBOT:
-		return min == TeamRole_NONE || min == TeamRole_RESTRICTEDBOT
+		switch min {
+		case TeamRole_NONE, TeamRole_RESTRICTEDBOT:
+			return true
+		default:
+			return false
+		}
+	case TeamRole_BOT:
+		switch min {
+		case TeamRole_NONE, TeamRole_RESTRICTEDBOT, TeamRole_BOT:
+			return true
+		default:
+			return false
+		}
 	default:
-		return int(r) >= int(min) || min == TeamRole_RESTRICTEDBOT
+		return (int(r) >= int(min) ||
+			min == TeamRole_BOT ||
+			min == TeamRole_RESTRICTEDBOT)
 	}
 }
```

Comment:
This method is getting tricky. How about using very-private int values like this? https://goplay.space/#-7jmf09gPbX

Reply:
yeah much better